### PR TITLE
Bug 1805388: Cleanup 4.4 deployment on downgrade to 4.3

### PR DESCRIPTION
**Pull request by marun**
4.3 uses a daemonset to manage apiserver pods, so the 4.4 deployment is likely to cause trouble if not removed.

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -229,3 +233,55 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
...
+		if deploy.ObjectMeta.DeletionTimestamp != nil {
+			// Done - deployment has been marked for deletion
+			return
+		}
+
+		// Check that the daemonset managing the apiserver pods has at last one available replica
+		ds, err := dsClient.Get(resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// No available replicas if the daemonset doesn't exist
+			return
```
Comment (sttts) [inline pkg/operator/starter.go]: same comment as for other PR: why not stop the loop?

Comment (marun) [reply pkg/operator/starter.go]: Please confer with @deads2k - this strategy is as per his requirement.

**Pull request merged**